packages/fx-core/templates/plugins/resource/keyvault/bicep/keyVaultProvision.template.bicep

```diff
@@ -6,7 +6,7 @@ var resourceBaseName = provisionParameters.resourceBaseName
 var keyVaultName = contains(provisionParameters, 'keyVaultName') ? provisionParameters['keyVaultName'] : '${resourceBaseName}' // Try to read name for Key Vault from parameters
 var tenantId = subscription().tenantId
 var m365ClientSecretName = 'm365ClientSecret' // Secret name of AAD app client secret
-var m365ClientSecret = provisionParameters['m365ClientSecret'] // Read AAD app client secret from parameters
+var m365ClientSecret = contains(provisionParameters, 'm365ClientSecret') ? provisionParameters['m365ClientSecret'] : '' // Try to read AAD app client secret from parameters
```

Comment:
since previous projects do not have such logic, will there be any compatibility problem on old projects?

Reply:
Most of previous projects will have aad plugin activated by default - only non SSO sample will have this problem. For non SSO sample, adding keyvault will also cause error, so I think it won't be a big problem.